internal/versions/version.go

```diff
@@ -0,0 +1,102 @@
+/*
+Copyright (c) Edgeless Systems GmbH
+
+SPDX-License-Identifier: AGPL-3.0-only
+*/
+
+package versions
+
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	"github.com/edgelesssys/constellation/v2/internal/constants"
+	"golang.org/x/mod/semver"
+)
+
+// Version represents a semantic version.
+type Version struct {
+	Major int
+	Minor int
+	Patch int
+}
+
+// NewVersion returns a Version from a string.
+func NewVersion(version string) (Version, error) {
+	// ensure that semver has "v" prefix
+	if !strings.HasPrefix(version, "v") {
+		version = "v" + version
+	}
+
+	if !semver.IsValid(version) {
+		return Version{}, fmt.Errorf("invalid semver: %s", version)
+	}
+
+	version = semver.Canonical(version)
+
+	var major, minor, patch int
+	_, err := fmt.Sscanf(version, "v%d.%d.%d", &major, &minor, &patch)
+	if err != nil {
+		return Version{}, fmt.Errorf("parsing semver parts: %w", err)
+	}
+
+	return Version{
+		Major: major,
+		Minor: minor,
+		Patch: patch,
+	}, nil
+}
+
+// String returns the string representation of the version.
+func (v Version) String() string {
+	return fmt.Sprintf("v%d.%d.%d", v.Major, v.Minor, v.Patch)
+}
+
+// Compare compares two versions. It relies on the semver.Compare function internally.
+func (v Version) Compare(other Version) int {
+	return semver.Compare(v.String(), other.String())
+}
+
+// CanUpgradeTo returns if a version can be upgraded to another version.
+// It checks if the version of other is greater than the current version and allows a drift of at most one minor version.
+func (v Version) CanUpgradeTo(other Version) bool {
```

Comment:
nit: I would prefer `IsUpgradeTo` as name, since version a either is an upgrade to version b or it is not. The act of upgrading is not done by the version object. What do you think?